Warn against callOp

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37797 Warn against callOp**

This is slow (see comment in code).
Not fixing this yet, but at least adding a warning so people are aware and don't add new call sites.

Differential Revision: [D21390364](https://our.internmc.facebook.com/intern/diff/D21390364/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21390364/)!

smessmer committed:
Warn against callOp

This is slow (see comment in code).
Not fixing this yet, but at least adding a warning so people are aware and don't add new call sites.

Differential Revision: [D21390364](https://our.internmc.facebook.com/intern/diff/D21390364/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21390364/)!

[ghstack-poisoned]

smessmer committed:
Update on "Warn against callOp"


This is slow (see comment in code).
Not fixing this yet, but at least adding a warning so people are aware and don't add new call sites.

Differential Revision: [D21390364](https://our.internmc.facebook.com/intern/diff/D21390364/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21390364/)!

[ghstack-poisoned]

ezyang commented on aten/src/ATen/native/quantized/README.md:
```diff
@@ -188,9 +188,13 @@ namespace at {
+  */
   template <class... Args>
-  inline std::vector<torch::IValue> callOp(
+  inline std::vector<torch::IValue> callOp_slow(
```
These are docs... can't you just update the docs so that they do things the right way?

smessmer replied:
ok I reverted the changes here for now because I'm only 99% (but not 100%) confident that op.callUnboxed works for all use cases where callOp worked.

ezyang commented on test/mobile/op_deps/main.cc:
```diff
@@ -13,7 +13,7 @@ int main() {
-  callOp("quantized::t_add", "", input, input, 1.0, 0);
-  callOp("quantized::t_add_relu", "", input, input, 1.0, 0);
+  callOp_slow("quantized::t_add", "", input, input, 1.0, 0);
+  callOp_slow("quantized::t_add_relu", "", input, input, 1.0, 0);
```
These calls don't matter, we're just trying to demonstrate that manually called ops get tracked. You can also just switch this to whatever the correct api is.

smessmer replied:
I could probably switch all of those to the correct API instead, which would be the better approach, but I only had time for this half-way solution. I wanted to at least make sure people don't continue to copy&paste this to new locations, it started with one implementation and now we have 3 `callOp` implementations.

bhosmer reviewed (changes requested):
@messmer just flagging that I'd like to chat re the naming here before you land it... will ping offline

smessmer committed:
Update on "Warn against callOp"


This is slow (see comment in code).
Not fixing this yet, but at least adding a warning so people are aware and don't add new call sites.

Differential Revision: [D21390364](https://our.internmc.facebook.com/intern/diff/D21390364/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21390364/)!

[ghstack-poisoned]

bhosmer reviewed (approved):
Excellent 😬

smessmer committed:
Update on "Warn against callOp"


This is slow (see comment in code).
Not fixing this yet, but at least adding a warning so people are aware and don't add new call sites.

Differential Revision: [D21390364](https://our.internmc.facebook.com/intern/diff/D21390364/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21390364/)!

[ghstack-poisoned]

smessmer committed:
Update on "Warn against callOp"


This is slow (see comment in code).
Not fixing this yet, but at least adding a warning so people are aware and don't add new call sites.

Differential Revision: [D21390364](https://our.internmc.facebook.com/intern/diff/D21390364/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21390364/)!

[ghstack-poisoned]

smessmer committed:
Update on "Warn against callOp"


This is slow (see comment in code).
Not fixing this yet, but at least adding a warning so people are aware and don't add new call sites.

Differential Revision: [D21390364](https://our.internmc.facebook.com/intern/diff/D21390364/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21390364/)!

[ghstack-poisoned]
